JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/agent/JPSAgent.java

```diff
@@ -54,5 +55,34 @@ public boolean validateInput(JSONObject requestParams) throws BadRequestExceptio
         }
         return true;
     }
-
+    
+    /**
+     * Execute a <a href="https://www.w3.org/TR/sparql11-query/">SPARQL Query</a> on the target resource 
+     * in the Knowledge Graph by calling the AccessAgent.
+     * @param targetResourceID	target namespace or IRI
+     * 							e.g. to access the Ontokin triple store
+     * 							both "ontokin" and "http://www.theworldavatar.com/kb/ontokin" are accepted.
+     * @param sparqlQuery		SPARQL query string
+     * @return 	the query result in the W3C Query result JSON format, 
+     * see <a href="https://www.w3.org/TR/sparql11-results-json/">Query Results JSON Format</a>
+     */
+    public String query(String targetResourceID, String sparqlQuery) {
+    	//pass the target resource ID directly as the targetUrl
+    	//both datasetUrl and targetUrl are not used by the AccessAgent for queries
+    	return AccessAgentCaller.query(null, targetResourceID, sparqlQuery);
```

Comment:
Maybe we could just have a second method for that in the `AccessAgentCaller` that does not need to pass the dataset/thatgetURL in the signature: like `query(String targetResourceID, String sparqlQuery)`.

Reply:
Added query(String targetResourceID, String sparqlQuery) and update(String targetResourceID, String sparqlUpdate) to AccessAgentCaller